src/credentials/sharedCredentialsValidation.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This module focuses on the validation of shared credentials properties
+ */
+
+import { localize } from 'vscode-nls'
+import { SectionName, SharedCredentialsKeys, StaticCredentialsProfileData } from './types'
+import { ToolkitError } from '../shared/errors'
+import { profileExists } from './sharedCredentials'
+
+/**
+ * The format validators for shared credentials keys.
+ *
+ * A format validator validates the format of the data,
+ * but not the validity of the content.
+ */
+export const CredentialsKeyFormatValidators = {
+    [SharedCredentialsKeys.AWS_ACCESS_KEY_ID]: getAccessKeyIdFormatError,
+    [SharedCredentialsKeys.AWS_SECRET_ACCESS_KEY]: getSecretAccessKeyFormatError,
+} as const
+
+/**
+ * Holds the error for each key of static credentials data,
+ * if it exists. This allows the user to get all the errors
+ * at once.
+ */
+export type StaticCredentialsErrorResult = {
+    [k in keyof StaticCredentialsProfileData]: string | undefined
+}
+
+export function getStaticCredentialsDataErrors(
+    data: StaticCredentialsProfileData
+): StaticCredentialsErrorResult | undefined {
+    const accessKeyIdError = CredentialsKeyFormatValidators[SharedCredentialsKeys.AWS_ACCESS_KEY_ID](
+        data[SharedCredentialsKeys.AWS_ACCESS_KEY_ID]
+    )
+    const secretAccessKeyError = CredentialsKeyFormatValidators[SharedCredentialsKeys.AWS_SECRET_ACCESS_KEY](
+        data[SharedCredentialsKeys.AWS_SECRET_ACCESS_KEY]
```

Comment:
There are 6 different symbols here with verbose names. At some point verbosity makes it hard to see patterns and compare nearby lines of code (in short: hard to read).

- Example is `SharedCredentialsKeys`: name seems over-specified. It's not likely that there will be other classes of "credentials keys", so it can be named `CredentialsKeys`.

```suggestion
    const keyError = CredentialsKeyFormatValidators[CredentialsKeys.AWS_SECRET_ACCESS_KEY](
        data[CredentialsKeys.AWS_SECRET_ACCESS_KEY]
```

Reply:
This makes sense to me, I'll refactor this in a future PR.

Would it even make sense to shorten it even more to something like `Credentials.AWS_SECRET_ACCESS_KEY`. I don't see scenario where we are only using `Credentials` it will always follow up with a property.